### ISSUE TITLE
Cannot kubectl exec thanos-compactor container

### ISSUE THREAD

**Issue by suxiaoxiaomm** [bitnami/containers#28661]
### Name and Version

docker.io/bitnami/thanos:0.29.0-scratch-r0

### What architecture are you using?

None

### What steps will reproduce the bug?

Hi,
Is there a way to do exec to the thanos-compactor container, I need to check the PVC mounted with this container.
Currently, I will get the below error message:
```console
$ kubectl exec -it observability-thanos-compactor-9d8cc7cd8-xdv8w -- shell
error: Internal error occurred: error executing command in container: failed to exec in container: failed to start exec "86dc5f72d91f433104bdf63c5d23257afaee347f65577d8379deeeb04a700cfe": OCI runtime exec failed: exec failed: container_linux.go:380: starting container process caused: exec: "shell": executable file not found in $PATH: unknow
````

Appreciate the help

### What is the expected behavior?

I can use a shell inside the thanos-compactor container.

### What do you see instead?

Got the below error:
```
error: Internal error occurred: error executing command in container: failed to exec in container: failed to start exec "86dc5f72d91f433104bdf63c5d23257afaee347f65577d8379deeeb04a700cfe": OCI runtime exec failed: exec failed: container_linux.go:380: starting container process caused: exec: "shell": executable file not found in $PATH: unknow
```

**Comment by carrodher** [comment]
Please note bitnami/thanos is a distroless container so there is no underlying distro, because of that you can't connect to this container using a shell

**Comment by suxiaoxiaomm** [comment]
Understand. Close the bug.

**Comment by danksim** [comment]
If I am running into an [overlap](https://thanos.io/tip/operating/troubleshooting.md/#overlaps) issue, what is a good way for me to debug? Is there a workaround to run the `thanos bucket` cmd since all components are running the `scratch` image? @carrodher

**Comment by andresbono** [comment]
Maybe you can combine [Debugging with an ephemeral debug container](https://kubernetes.io/docs/tasks/debug/debug-application/debug-running-pod/#ephemeral-container) and downloading the thanos binary to debug.

**Comment by danksim** [comment]
I've tried that (minus the thanos binary part) but there seems to be some hoops I need to jump through to mount the filesystem of the pod being debugged (compact component) to the ephemeral debugger pod. I also had to scale down the compact component in order for the debugger pod to bind to the PV since thanos PVCs' `accessModes` are all `ReadWriteOnce`. I am trying to figure out which labels I need to dedup/ignore instead of just deleting blocks.
